### PR TITLE
Error messages

### LINKED ISSUE #67: Error messages, I18N and Namespaces
Reform 2.0 does support lotus-validations and I will push towards making this the default implementation. Thanks for this. :heart: 

As discussed with Luca, I'm currently designing the process of generating an (i18nized) message string from an `Error` object. I do this in Reform first, and will then extract it to this gem.

Consider the following object for a nested field `band.label.name`. 

```ruby
#<Lotus::Validations::Error:0x007fb6d5b48aa0 @attribute_name="name", 
  @validation=:presence, @expected=true, @actual=nil, 
  @namespace="band.label", @attribute="name">]}
```

In order to translate this, I propose the following I18N paths (please comment!).

```
form.band.label.name.blank
form.band.name.blank
form.name.blank
```

Don't worry about the prefix `form.`, this could be anything and is just an example.

Another thing is: How to name the suffixes, e.g. `blank`? This has to be computed from `Error` by looking at `@validation` and `@actual`, so _we need to specify translation suffixes_ for validation/result tuples.

Any input?

### PR DESCRIPTION
**Pull request by jodosha**
WIP

Closes #67

### CODE REVIEW

File: lib/lotus/validations/error.rb [hunk 1]
```diff
@@ -90,6 +92,33 @@ def ==(other)
           other.expected   == expected   &&
           other.actual     == actual
       end
+
+      # FIXME for Luca - What to do when I18n isn't defined?
```
Comment (AlfonsoUceda) [inline lib/lotus/validations/error.rb]: but you're adding I18n to Gemfile, do we need check if I18n is defined?

Comment (jodosha) [inline lib/lotus/validations/error.rb]: @AlfonsoUceda Leave it to me ;) For now just assume it's available for that method :dancer:

Comment (AlfonsoUceda) [inline lib/lotus/validations/error.rb]: :+1:

**Comment by hlegius** [comment]
Great work! I am looking forward to it. :+1:

**Comment by matheusca** [comment]
Great @jodosha! :+1:

**Comment by DanielBlanco** [comment]
Any news on this PR?

**Comment by jodosha** [comment]
@DanielBlanco We're working on a more powerful backend, which already includes error messages. We're discussing in [chat](https://gitter.im/hanami/chat) what to do.

**Comment by jodosha** [comment]
Implemented by #100

**Pull request closed without merging**